### PR TITLE
various: use stream_read_file for loading files

### PR DESCRIPTION
**Pull request by na-na-hi**
This allows cleaning up some code duplication which can be served by `stream_read_file`.

### CODE REVIEW

File: stream/stream.h [hunk 1]
```diff
@@ -228,6 +228,8 @@ struct bstr stream_read_complete(struct stream *s, void *talloc_ctx,
                                  int max_size);
 struct bstr stream_read_file(const char *filename, void *talloc_ctx,
                              struct mpv_global *global, int max_size);
+struct bstr stream_read_file2(const char *filename, void *talloc_ctx,
```
Comment (kasper93) [inline stream/stream.h:231]: Could we have short docs describing different purposes on those functions? So we don't have to look at source? I understand one doesn't take flags.

**Review by sfan5 — APPROVED** [review]
looks good

File: stream/stream.h (hunk 1 above)
Comment (kasper93) [reply stream/stream.h:232]: Thanks.

**Review by kasper93 — APPROVED** [review]

**Comment by avih** [comment]
Hmm.. I noticed just now the JS change.

It's close, but not enough.

All the `js_*` APIs can throw, and if `js_pushlstring` throws (e.g. on OOM or whatever), then the bstr is leaked.

It should use the `af` argument as talloc context, and without final `talloc_free`, because this is an auto-free function, where the `af` argument is freed after the function exits or throws.

@na-na-hi apologies for not noticing this earlier. Mind posting a followup fix?

**Comment by kasper93** [comment]
With javascript plugin I get now:
```
[  0.025582]       file: Cannot open file '': Invalid argument
[  0.025655]     stream: Failed to open .
```
Is this expected?

@na-na-hi apologies for not noticing this earlier.

**Pull request merged**